File: 713-subarray-product-less-than-k/subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        if(k<=1) return 0;
        int n = nums.size();
        int i = 0;
        int res = 0;
        int prod = 1;
        for(int j = 0;j<n;j++){
            prod = prod*nums[j];
            while(prod>=k) {
                prod = prod/nums[i];
                i++;
            }
            res += (j-i+1);
        }
        return res;
    }
};

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int cnt = 0;

        int n = nums.length;

        int lastFound = -1;

        for(int i = 0;i<n;i++){
            long product = 1;
            for(int j = i;j<n;j++){
                product *= nums[j];
                if(product<k){
                    cnt++;
                    
                }
                else{
                   break;
                }
            }
        }

        return cnt;
    }
}